Example/Emission/UnroutedViewController.m

```diff
@@ -1,49 +1,106 @@
+
+  [request addValue:agentString forHTTPHeaderField:@"UserAgent"];
+  [request addValue:emission.configurationModule.userID forHTTPHeaderField:@"X-User-ID"];
+  [request addValue:emission.configurationModule.authenticationToken forHTTPHeaderField:@"X-Access-Token"];
+
+  [self.webView loadRequest:request];
 }
 
 - (void)viewDidLoad
 {
-  UILabel *label = [UILabel new];
-  label.text = self.route;
-  label.font = [UIFont serifFontWithSize:18];
-  label.textAlignment = NSTextAlignmentCenter;
-  label.numberOfLines = 0;
+  [super viewDidLoad];
+  WKWebViewConfiguration *config = [[WKWebViewConfiguration alloc] init];
+
+  CGRect deviceBounds = [UIScreen mainScreen].bounds;
+  WKWebView *webView = [[WKWebView alloc] initWithFrame:deviceBounds configuration:config];
 
-  self.view.backgroundColor = [UIColor colorWithWhite:0.9 alpha:1];
-  [self.view addSubview:label];
+  webView.frame = self.view.bounds;
+  webView.navigationDelegate = self;
+  [self.view addSubview:webView];
 
-  [label constrainWidthToView:self.view predicate:@"-40"];
-  [label alignCenterWithView:self.view];
+  _webView = webView;
 
-  UIButton *openInEigen = [UIButton buttonWithType:UIButtonTypeCustom];
-  openInEigen.titleLabel.font = [UIFont sansSerifFontWithSize:14];
-  [openInEigen setTitle:@"OPEN IN EIGEN" forState:UIControlStateNormal];
-  [self.view addSubview:openInEigen];
-  [openInEigen constrainTopSpaceToView:label predicate:@"20"];
-  [openInEigen alignCenterXWithView:self.view predicate:@"0"];
-  [openInEigen addTarget:self action:@selector(openInEigen) forControlEvents:UIControlEventTouchUpInside];
+  [self loadURL:self.initialURL];
 }
 
-- (void)openInEigen
+- (UIStatusBarStyle)preferredStatusBarStyle
 {
-  NSString *route = [NSString stringWithFormat:@"artsy:%@", self.route];
-  NSURL *url = [NSURL URLWithString:route];
-  [[UIApplication sharedApplication] openURL:url];
+  return UIStatusBarStyleLightContent;
 }
 
+- (void)viewWillLayoutSubviews
+{
+  [self.webView constrainTopSpaceToView:self.flk_topLayoutGuide predicate:@"0"];
+  [self.webView alignLeading:@"0" trailing:@"0" toView:self.view];
+  [self.webView alignBottomEdgeWithView:self.view predicate:@"0"];
+}
+
+- (UIStatusBarStyle)statusBarStyle
+{
+  return self.statusBarStyle;
+}
+
+
+#pragma mark - Properties
+
+#pragma mark WKWebViewDelegate
+
+- (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler;
+{
+  decisionHandler([self shouldLoadNavigationAction:navigationAction]);
+}
+
+- (WKNavigationActionPolicy)shouldLoadNavigationAction:(WKNavigationAction *)navigationAction;
+{
+  if (navigationAction.navigationType == WKNavigationTypeLinkActivated) {
+    NSString *route = navigationAction.request.URL.absoluteString;
```

Comment:
This should probably the instance method `-[NSURL path]` (instead of `absoluteString`), since the existing Switchboard code assumes relative paths (for artist, gene, etc.).

Reply:
This would break something like the invoices, as you need to keep the URL too.

Reply:
The suggestion of using `-[NSURL path]` was not a comprehensive one, but more to point out that this is currently not doing what readers may expect it to do.